osquery/dispatcher/scheduler.cpp

```diff
@@ -150,38 +151,64 @@ inline void launchQuery(const std::string& name, const ScheduledQuery& query) {
 }
 
 void SchedulerRunner::start() {
-  // Start the counter at the second.
-  auto i = osquery::getUnixTime();
-  for (; (timeout_ == 0) || (i <= timeout_); ++i) {
-    Config::get().scheduledQueries(
-        ([&i](const std::string& name, const ScheduledQuery& query) {
-          if (query.splayed_interval > 0 && i % query.splayed_interval == 0) {
-            TablePlugin::kCacheInterval = query.splayed_interval;
-            TablePlugin::kCacheStep = i;
-            launchQuery(name, query);
-          }
-        }));
+  auto tasks = ScheduledTaskQueue{};
+  Config::get().scheduledQueries(
+      [&tasks](const std::string& name, const ScheduledQuery& query) {
+        if (query.splayed_interval > 0) {
+          tasks.add(
+              [name, &query](const auto start_time) {
+                TablePlugin::kCacheInterval = query.splayed_interval;
+                TablePlugin::kCacheStep = start_time;
+                launchQuery(name, query);
+                return query.splayed_interval;
+              },
+              query.splayed_interval);
+        }
+      });
```

Comment:
Via config refresh new queries can be pushed. I dont see if you have made arrangements for scheduling new queries.

Reply:
Right, this needs to accommodate schedule updates. Those updates can be intense, merging different sources together into a new schedule. The configuration does not calculate a delta of additions / removals so this might not feel good. You'll have to iterate the whole schedule and add the new tasks by query name, and remove old tasks no longer in the schedule.

Reply:
I didn't find where exactly scheduled queries could be added to the config. Is it already implemented, or it is just a planned feature?